lapce-data/src/find.rs

```diff
@@ -278,48 +267,50 @@ impl Find {
                 ) {
                     let end = find_cursor.pos();
 
-                    if self.whole_words
-                        && !self.is_matching_whole_words(text, start, end)
+                    if self.whole_words && !is_matching_whole_words(text, start, end)
                     {
                         raw_lines = text.lines_raw(find_cursor.pos()..text.len());
                         continue;
                     }
+
                     raw_lines = text.lines_raw(find_cursor.pos()..text.len());
 
                     if start > offset {
                         regions.push((start, end));
                     }
                 }
+
                 if !regions.is_empty() {
                     return Some(regions[regions.len() - 1]);
                 }
             }
         }
+
         None
     }
 
-    /// Execute the search on the provided text in the range provided by `start` and `end`.
+    /// Performs a search on the specified text within the range defined by `start` and `end`.
     pub fn update_find(
         &mut self,
         text: &Rope,
         start: usize,
         end: usize,
         include_slop: bool,
     ) {
-        if self.search_string.is_none() {
+        let search_string = if let Some(s) = self.search_string.as_ref() {
+            s
+        } else {
             return;
-        }
+        };
 
```

Comment:
```rs
        let Some(search_string) = self.search_string.as_ref() else {
            return;
        };
```

Reply:
I'm confused a bit. Is that code syntactically correct? Is `x else { return; }` possible?

Reply:
https://doc.rust-lang.org/stable/rust-by-example/flow_control/let_else.html

Reply:
Oh! It's been available since `rust 1.65` :smile: